keps/sig-apps/0003-kubernetes-application-api.md

```diff
@@ -0,0 +1,1120 @@
+       spec:
+         accessModes: [ "ReadWriteOnce" ]
+         resources:
+           requests:
+             storage: 250Gi
+--- 
+apiVersion: v1
+Kind: Application
+metadata:
+  name: wordpress
+spec:
+  type: wordpress
+  version: 1.0
+  url: 
+  selector:
+    matchLables:
+      app: wordpress
+  components:
+    - wordpress-hsvc:
+        version: v1
+        kind: Service
+    - wordpress-svc:
+        version: v1
+        kind: Service
+    - wordpress-webserver:
+        group: apps
+        version: v1
+        kind: StatefulSet
+   dependencies:
+    - mysql
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress-svc
+  labels:
+    app: wordpress
+    component: wordpress-svc
+  annotations:
+      kubernetes.io/application: wordpress
+spec:
+  ports:
+    - port: 80
+  selector:
+    app: wordpress
+    component: wordpress-svc
+  type: LoadBalancer
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress-hsvc
+  labels:
+    app: wordpress
+    component: wordpress-hsvc
+  annotations:
+      kubernetes.io/application: wordpress
+spec:
+  ports:
+    - port: 3306
+  selector:
+    app: wordpress
+    component: wordpress-hsvc
+  clusterIP: None
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: wordpress-webserver
+  labels:
+    app: wordpress
+    component: wordpress-webserver
+  annotations:
+      kubernetes.io/application: wordpress
+spec:
+  replicas: 1
+  service: wordpress-hsvc
+  selector:
+    matchLabels:
+      app: wordpress
+      component: wordpress-webserver
+  template:
+    metadata:
+      labels:
+        app: wordpress
+        tier: frontend
+    spec:
+      containers:
+      - image: wordpress:4.8-apache
+        name: wordpress
+        env:
+        - name: WORDPRESS_DB_HOST
+          value: wordpress-mysql
+        - name: WORDPRESS_DB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mysql-pass
+              key: password
+        ports:
+        - containerPort: 80
+          name: wordpress
+        volumeMounts:
+        - name: wordpress-persistent-storage
+          mountPath: /var/www/html
+  volumeClaimTemplates:
+       - metadata:
+           name: mysql-persistent-storage
+         spec:
+           accessModes: [ "ReadWriteOnce" ]
+           resources:
+             requests:
+               storage: 250Gi
+```
+
+Above we generate two Applications, one for the MySQL RDBMs and another for the WordPress web server. Each of these 
+Applications can be managed independently. For instance, if the WordPress web server is deleted, the components of the 
+MySQL RDBMS will not be affected. However, the dependency between the two workloads is made explicit in the 
+`Spec.Dependencies` of the `wordpress` Application. Tools and UIs that examine the `wordpress` application are made 
+aware of its dependency on the `mysql` Application, and they can surface the dependency to users.
+
+#### Selecting Components
```

Comment:
There's no real discussion about RBAC requirements and the Application type. I think it needs some explicit discussion. Is the controller acting "on behalf of" a user when it works on these components? If I have perms to delete an app, is that sufficient to delete all the app's components, or do I also have to have RBAC permissions on each object that is to be deleted?